R/Replace_NAs.R

```diff
@@ -81,16 +99,18 @@ ReplaceNAs <- function(data,pct.NA.row=70,pct.NA.column=80) {
 
         if (do_knn==TRUE){
             print("The first round of missing values imputation failed or resulted in negative values. Running additional missing value imputation using impute.knn. Rows with more than 50% missing values will be imputed using the overall mean per sample.")
-            data <- impute.knn(as.matrix(data), rowmax = 0.5)   ##This might generate "Error: C stack usage  xxx is too close to the limit" (probably the problem of insufficient RAM)
-            data <- data.frame(data$data)
+            imputed_values <- impute.knn(as.matrix(assay(SE_subset_rows_columns_rows)), rowmax = 0.5)   ##This might generate "Error: C stack usage  xxx is too close to the limit" (probably the problem of insufficient RAM)
+            assay(SE_subset_rows_columns_rows, withDimnames=FALSE) <- data.frame(imputed_values$data)
         } else {
             print("Missing value estimation using local least squares finished successfully.")
-            data <- data.lls
+            assay(SE_subset_rows_columns_rows, withDimnames=FALSE) <- data.lls
         }
-        rownames(data) <- varnames
+
+        final_output<-SE_subset_rows_columns_rows
     }
 
-    return(data)
-}
+    final_output<-SE_subset_rows_columns
```

Comment:
I don't think this assignment is necessary

Reply:
I have put there this assignment because the final result could have 2 forms depending on the if condition.
Or am I missing something?

Reply:
I don't think this is the case as the code is now, I see:

```
    if (TRUE %in% still.NA) {
         ...
        final_output<-SE_subset_rows_columns_rows
    }

    final_output<-SE_subset_rows_columns

    return(final_output)
```

here we are always returning `final_output` which is always assigned to `SE_subset_rows_columns` (the assignment is not conditional - happens every time)

what you maybe want is then

```
    if (TRUE %in% still.NA) {
         ...
        final_output<-SE_subset_rows_columns_rows
    } else {
        final_output<-SE_subset_rows_columns
    }

    return(final_output)
```

but you might as well

```
    if (TRUE %in% still.NA) {
         ...
        return(SE_subset_rows_columns_rows)
    } else {
        return(SE_subset_rows_columns)
    }
```

Reply:
cool, thanks